scripts/readelf.py

```diff
@@ -1332,6 +1335,58 @@ def _dump_debug_frames_interp(self):
                     self._dwarfinfo.debug_frame_sec,
                     self._dwarfinfo.CFI_entries())
 
+    def _dump_debug_locations(self):
+        """ Dump the location lists from .debug_location section
+        """
+        def _get_cu_base(cu):
+            top_die = cu.get_top_DIE()
+            attr = top_die.attributes
+            if 'DW_AT_low_pc' in attr:
+                return attr['DW_AT_low_pc'].value
+            elif 'DW_AT_entry_pc' in attr:
+                return attr['DW_AT_entry_pc'].value
+            else:
+                raise ValueError("Can't find the base IP (low_pc) for a CU")                    
+
+        di = self._dwarfinfo
+        loc_lists = di.location_lists()
```

Comment:
could you do something like 

```
loc_lists = list(di.location_lists() or [])
```
This would save a bunch of lines of logic

Reply:
Readelf distinguishes between the cases of no locations section and empty locations section - the latter produces a message.

Reply:
OK, early returns are better then. Use `list` anyway instead of list comprehension here. Also, please add a comment explaining that we want to distinguish between these cases. Thanks!